SDK-v4/gdpr/app/src/main/java/com/localytics/gdpr/LocalyticsGDPRWrapper.java

```diff
@@ -0,0 +1,246 @@
+package com.localytics.gdpr;
+
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.content.Intent;
+
+import com.localytics.android.Localytics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * This class acts as a wrapper around the Localytics implementation to ensure that no data can be
+ * passed to the Localytics system if a user has opted out of data collection.
+ *
+ * This implementation expects that every time the app is created, the opted in state of the user is
+ * checked against a backend source of truth.  As a result, the opted in state will be stored as an
+ * instance variable, and persistent storage will not be required.
+ */
+public class LocalyticsGDPRWrapper {
+
+    private static LocalyticsGDPRWrapper INSTANCE = null;
+
+    /**
+     * A boolean representing the user's current opt out status
+     */
+    private boolean optedOut = false;
+
+    /**
+     * A boolean representing whether or not datapoints can immediately be passed to Localytics.
+     * This value should be true when the current user's opt in status is unknown by the app.
+     */
+    private boolean queueDatapoints = false;
+
+    /**
+     * A queue of event datapoints that have been triggered by the app while the user's opt in state is unknown
+     */
+    private List<Runnable> eventQueue = new ArrayList<>();
+
+    static LocalyticsGDPRWrapper getInstance() {
```

Comment:
Since we expect fetching from a backend server, it should be clear what happens to things before the status is determined. May be we need a optedOut not determined. Assuming that optedOut is false by default is based if this is for GDPR.

Reply:
This app makes sure that on app launch everything is suppressed (in the `Application` class).